src/components/Activities/EditActivityModal.tsx

```diff
@@ -0,0 +1,247 @@
+}
+
+export const EditActivityModal: FunctionComponent<IEditActivityModalProps> = (props) => {
+
+  const weekStart = DateUtilities.getStartOfWeek(props.activity?.WeekOf);
+  const [activity, setActivity] = useState<IActivity>({ ...props.activity });
+  const [validated, setValidated] = useState<boolean>(false);
+  const [selectedDate, setSelectedDate] = useState<Date>(DateUtilities.momentToDate(weekStart));
+  const [highlightDates, setHighlightDates] = useState<Date[]>(DateUtilities.getWeek(weekStart));
+  const [datePickerOpen, setDatePickerOpen] = useState<boolean>(false);
+
+  const convertOPRsToPersonas = (OPRs?: UserList) => {
+    let personas: { text: string; imageInitials: string; SPUserId: string; Email: any; }[] = [];
+    if (OPRs && OPRs.results) {
+      personas = OPRs.results.map(OPR => {
+        return {
+          text: OPR.Title,
+          imageInitials: OPR.Title.substr(OPR.Title.indexOf(' ') + 1, 1) + OPR.Title.substr(0, 1),
+          SPUserId: OPR.Id,
+          Email: OPR.Email
+        }
+      })
+    }
+    return personas;
+  }
+
+  const handleOpen = () => {
+    let weekStart = DateUtilities.getStartOfWeek(props.activity.WeekOf);
+    // Deep copy activity
+    let editActivity = { ...props.activity };
+    editActivity.OPRs = {
+      results: props.activity.OPRs ? [...props.activity.OPRs.results] : []
+    };
+    //editActivity.InputWeekOf = editActivity.WeekOf.split('T', 1)[0];
+    setActivity(editActivity);
+    setValidated(false);
+    setSelectedDate(DateUtilities.momentToDate(weekStart));
+    setHighlightDates(DateUtilities.getWeek(weekStart));
+  }
+
+  // Syncs fields between react state and form
+  const updateActivity = (newValue: any, fieldUpdating: string): void => {
+    setActivity({ ...activity, [fieldUpdating]: newValue });
+  }
+
+  const updateOPRs = (value: SPPersona[]) => {
+    const newActivity: IActivity = { ...activity };
+    newActivity.OPRs = {
+      results: value.map((newOPR) => {
+        return {
+          Id: newOPR.SPUserId ? newOPR.SPUserId : '-1',
+          Title: newOPR.text ? newOPR.text : '',
+          Email: newOPR.Email
+        }
+      })
+    }
+    setActivity(newActivity);
+  }
+
+  const isOPRInvalid = () => {
+    return activity.OPRs && activity.OPRs.results && activity.OPRs.results.length ? false : true;
+  }
+
+  const validateActivity = () => {
+    const form = document.getElementById("EditActivityModal");
+    if (form instanceof HTMLSelectElement && form.checkValidity() === false || isOPRInvalid()) {
+      setValidated(true);
+    } else {
+      props.submitEditActivity(activity);
+      setValidated(false);
+    }
+  }
+
+  const onDateChange = (date: Date | Moment | string | null) => {
+    let selectedDate = DateUtilities.getStartOfWeek(date);
+    let highlightDates = DateUtilities.getWeek(selectedDate);
+    //let newActivity = { ...activity };
+    //activity.InputWeekOf = selectedDate;
+    //setActivity(newActivity);
```

Comment:
don't need these comments

Reply:
Forgot about those lol, I commented them out because I wasn't sure if getting rid of InputWeekOf would break anything